lib/formatters.js

```diff
@@ -1,27 +1,50 @@
 'use strict';
 
 const Promise = require('bluebird');
+const { DataEntity } = require('@terascope/utils');
 
 // No parsing, leaving to reader or a downstream op.
-function pass(data) {
-    return data;
+function raw(data, logger, metadata, opConfig) {
+    return Promise.map(data, (record) => {
+        try {
+            return DataEntity.make(
+                { data: record },
+                metadata
+            );
+        } catch (err) {
+            if (opConfig._dead_letter_action === 'log') {
+                logger.error('Bad record');
```

Comment:
This might appear better in the logs

```suggestion
                logger.error(err, 'Bad record', record);
```

Reply:
Should it always log bad records, or should logging the data be optional like the [hdfs appender](https://github.com/terascope/hdfs-assets/blob/master/asset/hdfs_append/index.js#L184)?

Reply:
It should because logging is opt-in with `_dead_letter_action`, which means you'd don't need `log_data_on_error`

Reply:
Although this implementation cannot include the dead letter queue which makes this potentially more limited.

Reply:
Oh, ya, that should've been more obvious to me.. Fixed